core/templates/pages/exploration-player-page/layout-directives/learner-view-info.component.ts

```diff
@@ -119,8 +124,8 @@ export class LearnerViewInfoComponent {
   }
 
   getTopicUrl(): string {
-    let topicUrlFragment: string;
-    let classroomUrlFragment: string;
+    let topicUrlFragment!: string;
+    let classroomUrlFragment!: string;
```

Comment:
I think we should explicitly assign and allow `null` here.

Reply:
No need, we are using "try to get values and catch for unwanted results" to get these properties. And further checking through if conditions. Also urlServices( who's value we are using in above properties)  never return null.

Reply:
But if the try-catch fails below these will not be set and wouldn't it be better to have them set to `null`?

Reply:
Done